cpp/daal/src/algorithms/kernel_config.h

```diff
@@ -24,6 +25,9 @@
 #ifndef __KERNEL_CONFIG_H__
 #define __KERNEL_CONFIG_H__
 
-#include "src/algorithms/kernel_inst_x86.h"
-
+#ifdef __ARM_ARCH
+    #include "src/algorithms/kernel_inst_arm.h"
```

Comment:
I would prefer new files for an arch to go into a sub-folder. Otherwise, if more architectures are supported, the directory will get cluttered.

Reply:
yes, and also we have to have rules that can guide ownership rules. So we can't use flat files schema